lib/chf/data_fixes/util.rb

```diff
@@ -0,0 +1,29 @@
+module CHF
+  module DataFixes
+    module Util
+
+      # Generic utility code for updating all GenericWorks, with a progress
+      # bar.  Caller must yield a block that takes a single GeneircWork, possibly
+      # updates it in place (without #save'ing), and returns true only if it
+      # was updated.
+      def self.update_works
+        works_updated = []
+        progress_bar = ProgressBar.create(total: GenericWork.count, format: "|%B| %p%% %e %t")
+
+        GenericWork.find_each do |w|
+          updated = yield(w)
```

Comment:
Yep! It was always my plan, but I wanted to wait until I had 2-3 examples before extracting anything out, to make sure my abstraction met the actual design needs and wasn't premature over-engineering.